File: 250624/최고의 33위치/best-place-of-33.java
import java.util.Scanner;

public class Main {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        int n = sc.nextInt();
        int[][] grid = new int[n][n];
        for (int i = 0; i < n; i++) {
            for (int j = 0; j < n; j++) {
                grid[i][j] = sc.nextInt();
            }
        }
        sc.close();

        int ans = 0;
      
        for (int i = 0; i <= n - 3; i++) {
            for (int j = 0; j <= n - 3; j++) {
                int cnt = 0;
                
                for (int di = 0; di < 3; di++) {
                    for (int dj = 0; dj < 3; dj++) {
                        if (grid[i + di][j + dj] == 1) {
                            cnt++;
                        }
                    }
                }
                ans = Math.max(ans, cnt);
            }
        }

        System.out.println(ans);
    }
}

File: 251021/최고의 33위치/best-place-of-33.swift
import Foundation

guard let nLine = readLine(), let n = Int(nLine) else { fatalError() }
var grid = [[Int]]()

for _ in 0..<n {
    let row = readLine()!.split(separator: " ").map { Int($0)! }
    grid.append(row)
}
var ans = 0 

func check(_ x: Int, _ y: Int) -> Int {
    var cnt = 0 
    for i in x...x+2 {
        for j in y...y+2 {
            if grid[i][j] == 1 {
                cnt += 1 
            }
        }
    }
    return cnt 
}

for i in 0...n-3 {
    for j in 0...n-3 {
        ans = max(ans, check(i,j))
    }
}

print(ans)

